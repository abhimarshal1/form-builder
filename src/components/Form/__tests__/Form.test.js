import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

// Components
import { Form } from '..';

const renderer = new ShallowRenderer();

describe('Form', () => {
  test('shallow render & match the snapshot', () => {
    renderer.render(
      <Form
        onSubmit={jest.fn()}
        items={[
          {
            key: 'username',
            type: 'string',
            label: 'Username',
            placeholder: 'Enter username',
            required: true,
          },
          {
            key: 'password',
            type: 'password',
            label: 'Password',
            placeholder: 'Enter Password',
            required: true,
          },
        ]}
      />,
    );

    const output = renderer.getRenderOutput();
    expect(output).toMatchSnapshot();
  });
});
