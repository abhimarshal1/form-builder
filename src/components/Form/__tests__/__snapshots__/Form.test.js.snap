// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form shallow render & match the snapshot 1`] = `
<Form
  as="form"
  id="form-mduPSC58tn"
  onSubmit={[Function]}
>
  <FormInput
    as={[Function]}
    control={[Function]}
    label="Username"
    name="username"
    onChange={[Function]}
    placeholder="Enter username"
    required={true}
    type="text"
    value=""
  />
  <FormInput
    as={[Function]}
    control={[Function]}
    label="Password"
    name="password"
    onChange={[Function]}
    placeholder="Enter Password"
    required={true}
    type="password"
    value=""
  />
  <Button
    as="button"
    primary={true}
    type="submit"
  >
    SUBMIT
  </Button>
</Form>
`;
